starlight/views.py

```diff
@@ -6,23 +6,70 @@
 # License, or (at your option) any later version.
 
 from django.contrib.auth.views import logout
+from django.http import HttpResponseRedirect
 from django.shortcuts import render
 from django.views.decorators.csrf import csrf_protect
 
-from starlight.models import Skill, Employee
+from starlight.forms import EditForm, CompetencyForm
+from starlight.models import Skill, Employee, Competency
 
 
 def home(request):
     skills = Skill.objects.order_by('name')
     employees = Employee.objects.all()
-    return render(request, 'views/home.html', {'skills': skills, 'employees': employees, 'viewname': 'home'})
+    return render(request, 'views/home.html', {'skills': skills, 'employees': employees, 'viewgroup': 'home'})
 
 
 @csrf_protect
 def login(request):
     return render(request, 'login/login.html')
 
 
+def profile(request, id):
+    employee = Employee.objects.get(pk=id)
+    return render(request, 'profile_views/profile.html', {'employee': employee, 'viewgroup': 'profile'})
+
+
+def edit_competency(request, employee, id):
+    competency = Competency.objects.get(pk=id)
+    employee = Employee.objects.get(pk=employee)
+    skill = competency.skill
+    if request.method == 'POST':
+        form = EditForm(request.POST)
+        if form.is_valid():
+            interest = form.cleaned_data['interest']
+            experience = form.cleaned_data['experience']
+            competency_new, created = Competency.objects.get_or_create(skill=skill, interest=interest, experience=experience)
+            employee.competencies.remove(competency)
+            employee.competencies.add(competency_new)
+            employee.save()
+            return HttpResponseRedirect('/{}/profile/'.format(employee.id))
+
+    else:
+        form = EditForm()
+
+    return render(request, 'profile_views/edit_competency.html', {'form': form, 'competency': competency, 'viewgroup': 'profile'})
+
+
+def new_competency(request, employee):
+    employee = Employee.objects.get(pk=employee)
+    if request.method == 'POST':
+        form = CompetencyForm(request.POST, employee=employee)
+        if form.is_valid():
+            skill = form.cleaned_data['skill']
+            interest = form.cleaned_data['interest']
+            experience = form.cleaned_data['experience']
+            competency_new, created = Competency.objects.get_or_create(skill=skill, interest=interest, experience=experience)
```

Comment:
If a second employee has the same skill, interest and experience level, I believe this will add the same `Competency` record to a different employee and remove the association from the original. So make sure to add an employee filter to your call to `get_or_create` :)

Reply:
Also: ideally, if a duplicate is found on saving (i.e. the user already has an entry for that skill), the old one should be overwritten by what the user is trying to add, perhaps with a message that an old entry for that skill was overwritten.